classes/core/PKPRoutingProvider.php

```diff
@@ -0,0 +1,181 @@
+
+        $router->pattern('contextPath', '(.*?)');
+        $router->pattern('version', '(.*?)');
+    }
+
+    protected function registerResponseBindings(): void
+    {
+        $container = PKPContainer::getInstance();
+
+        $container->bind(\Illuminate\Routing\RouteCollectionInterface::class, \Illuminate\Routing\RouteCollection::class);
+        $container->bind(
+            \Illuminate\View\ViewFinderInterface::class, 
+            fn ($app) => new \Illuminate\View\FileViewFinder(app(\Illuminate\Filesystem\Filesystem::class), [])
+        );
+        $container->bind(\Illuminate\Contracts\View\Factory::class, \Illuminate\View\Factory::class);
+        $container->bind(\Illuminate\Contracts\Routing\ResponseFactory::class, \Illuminate\Routing\ResponseFactory::class);
+    }
+
+    /**
+     * Register the URL generator service.
+     *
+     * @return void
+     */
+    protected function registerUrlGenerator()
+    {
+        $this->app->singleton('url', function ($app) {
+            $routes = $app['router']->getRoutes();
+
+            // The URL generator needs the route collection that exists on the router.
+            // Keep in mind this is an object, so we're passing by references here
+            // and all the registered routes will be available to the generator.
+            $app->instance('routes', $routes);
+
+            return new UrlGenerator(
+                $routes, 
+                Request::capture(), 
+                $app['config']['app.asset_url']
+            );
+        });
+
+        $this->app->singleton(UrlGeneratorContract::class, function($app) {
+            return $app['url'];
+        });
+
+        $this->app->extend('url', function (UrlGeneratorContract|UrlGenerator $url, $app) {
+            // Next we will set a few service resolvers on the URL generator so it can
+            // get the information it needs to function. This just provides some of
+            // the convenience features to this URL generator like "signed" URLs.
+            $url->setSessionResolver(function () {
+                return $this->app['session'] ?? null;
+            });
+
+            $url->setKeyResolver(function () {
+                return $this->app->make('config')->get('app.key');
+            });
+
+            // If the route collection is "rebound", for example, when the routes stay
+            // cached for the application, we will need to rebind the routes on the
+            // URL generator instance so it has the latest version of the routes.
+            $app->rebinding('routes', function ($app, $routes) {
+                $app['url']->setRoutes($routes);
+            });
+
+            return $url;
+        });
+    }
+
+    /**
+     * Register a binding for the PSR-7 request implementation.
+     *
+     * @return void
+     *
+     * @throws \Illuminate\Contracts\Container\BindingResolutionException
+     */
+    protected function registerPsrRequest()
```

Comment:
This looks like it's just copied in from the parent class; if so, please remove it. Anyway, the packages that are being checked for in our package list.

Reply:
see comment at https://github.com/pkp/pkp-lib/pull/9176#discussion_r1288266039